streams/src/main/java/org/apache/kafka/streams/state/internals/metrics/StateStoreMetrics.java

```diff
@@ -0,0 +1,572 @@
+    private static final String SUPPRESSION_BUFFER_SIZE_MAX_DESCRIPTION =
+        MAX_DESCRIPTION_PREFIX + SUPPRESSION_BUFFER_SIZE_DESCRIPTION;
+
+    private static final String EXPIRED_WINDOW_RECORD_DROP = "expired-window-record-drop";
+    private static final String EXPIRED_WINDOW_RECORD_DROP_DESCRIPTION = "dropped records due to an expired window";
+    private static final String EXPIRED_WINDOW_RECORD_DROP_TOTAL_DESCRIPTION =
+        TOTAL_DESCRIPTION + EXPIRED_WINDOW_RECORD_DROP_DESCRIPTION;
+    private static final String EXPIRED_WINDOW_RECORD_DROP_RATE_DESCRIPTION =
+        RATE_DESCRIPTION_PREFIX + EXPIRED_WINDOW_RECORD_DROP_DESCRIPTION + RATE_DESCRIPTION_SUFFIX;
+
+
+    public static Sensor putSensor(final String threadId,
+                                   final String taskId,
+                                   final String storeType,
+                                   final String storeName,
+                                   final StreamsMetricsImpl streamsMetrics) {
+        return throughputAndLatencySensor(
+            threadId,
+            taskId,
+            storeType,
+            storeName,
+            PUT,
+            PUT_RATE_DESCRIPTION,
+            PUT_TOTAL_DESCRIPTION,
+            PUT_AVG_LATENCY_DESCRIPTION,
+            PUT_MAX_LATENCY_DESCRIPTION,
+            RecordingLevel.DEBUG,
+            streamsMetrics
+        );
+    }
+
+    public static Sensor putIfAbsentSensor(final String threadId,
+                                           final String taskId,
+                                           final String storeType,
+                                           final String storeName,
+                                           final StreamsMetricsImpl streamsMetrics) {
+        return throughputAndLatencySensor(
+            threadId,
+            taskId,
+            storeType,
+            storeName,
+            PUT_IF_ABSENT,
+            PUT_IF_ABSENT_RATE_DESCRIPTION,
+            PUT_IF_ABSENT_TOTAL_DESCRIPTION,
+            PUT_IF_ABSENT_AVG_LATENCY_DESCRIPTION,
+            PUT_IF_ABSENT_MAX_LATENCY_DESCRIPTION,
+            RecordingLevel.DEBUG,
+            streamsMetrics
+        );
+    }
+
+    public static Sensor putAllSensor(final String threadId,
+                                      final String taskId,
+                                      final String storeType,
+                                      final String storeName,
+                                      final StreamsMetricsImpl streamsMetrics) {
+        return throughputAndLatencySensor(
+            threadId,
+            taskId,
+            storeType,
+            storeName,
+            PUT_ALL,
+            PUT_ALL_RATE_DESCRIPTION,
+            PUT_ALL_TOTAL_DESCRIPTION,
+            PUT_ALL_AVG_LATENCY_DESCRIPTION,
+            PUT_ALL_MAX_LATENCY_DESCRIPTION,
+            RecordingLevel.DEBUG,
+            streamsMetrics
+        );
+    }
+
+    public static Sensor getSensor(final String threadId,
```

Comment:
This is a meta comment: in KIP-439 we also discussed about fixing some metric names which are collapsed and messed with the functionality names, and back then the proposal is:

```
value-by-key
value-by-key-and-time
range-by-key
range-by-key-range
range-by-time-range
range-by-key-and-time-range
range-by-key-range-and-time-range
range-all
```

Besides the first two, all others are mainly used in IQ only and not in stream processor at all, which means that only the first two are commonly recorded in practice. So I'm thinking maybe we keep the compatibility for the first two while renaming others:

```
get (in kv-store, it means value-by-key, in window-store, it means value-by-key-time)
range-by-key
range-by-key-range
range-by-time-range
range-by-key-and-time-range
range-by-key-range-and-time-range
range-all
```

WDYT cc @vvcephei @mjsax who have participated in the discussion of KIP-439 as well.

Reply:
Hey @guozhangwang . This seems good to me. It shouldn't be too ambiguous either way, and the documentation would have to explain it, anyway.

Reply:
Ack. I am ok with this suggestions -- will keep it in mind for KIP-439.

Btw: it this a change for KIP-444 what needs to be communicated as follow up to the already accepted KIP?

Reply:
I think KIP-444 just keep the metrics names on state stores as is -- we will do the renaming on KIP-439 and communicate that in that KIP.